tests/devices/qubit/test_simulate.py

```diff
@@ -1178,3 +1198,371 @@ def test_qinfo_tf(self):
+            tree_stack.results_1[depth] = [1] * counts_1
+            tree_stack.states[depth] = [np.sqrt(counts_0), np.sqrt(counts_1)]
+            assert tree_stack.is_full(depth)
+
+            assert tree_stack.counts[depth] == list(
+                samples_to_counts(
+                    np.array(tree_stack.results_0[depth] + tree_stack.results_1[depth])
+                ).values()
+            )
+            assert tree_stack.probs[depth] == list(
+                counts_to_probs(dict(zip([0, 1], tree_stack.counts[depth]))).values()
+            )
+
+            state_vec = np.array(tree_stack.states[depth]).T
+            state_vec /= np.linalg.norm(tree_stack.states[depth])
+            meas, meas_r = qml.measure(0), qml.measure(0, reset=True)
+            assert np.allclose(branch_state(state_vec, 0, meas.measurements[0]), [1.0, 0.0])
+            assert np.allclose(branch_state(state_vec, 1, meas_r.measurements[0]), [1.0, 0.0])
+
+            tree_stack.prune(depth)
+            assert tree_stack.any_is_empty(depth)
+
+
+class TestMidMeasurements:
+    """Tests for simulating scripts with mid-circuit measurements using the ``simulate_tree_mcm``."""
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("val", [0, 1])
+    def test_basic_mid_meas_circuit(self, val):
+        """Test execution with a basic circuit with mid-circuit measurements."""
+        qs = qml.tape.QuantumScript(
+            [qml.Hadamard(0), qml.CNOT([0, 1]), qml.measurements.MidMeasureMP(0, postselect=val)],
+            [qml.expval(qml.X(0)), qml.expval(qml.Z(0))],
+        )
+        result = simulate_tree_mcm(qs)
+        assert result == (0, (-1.0) ** val)
+
+    @pytest.mark.unit
+    def test_basic_mid_meas_circuit_with_reset(self):
+        """Test execution with a basic circuit with mid-circuit measurements."""
+        qs = qml.tape.QuantumScript(
+            [
+                qml.Hadamard(0),
+                qml.Hadamard(1),
+                qml.CNOT([0, 1]),
+                (m0 := qml.measure(0, reset=True)).measurements[0],
+                qml.Hadamard(0),
+                qml.CNOT([1, 0]),
+            ],  # equivalent to a circuit that gives equiprobable basis states
+            [qml.probs(op=m0), qml.probs(op=qml.Z(0)), qml.probs(op=qml.Z(1))],
+        )
+        result = simulate_tree_mcm(qs)
+        assert qml.math.allclose(result, qml.math.array([0.5, 0.5]))
+
+    # pylint: disable=too-many-arguments
+    @pytest.mark.unit
+    @pytest.mark.parametrize("shots", [None, 5500])
+    @pytest.mark.parametrize("postselect", [None, 0])
+    @pytest.mark.parametrize("reset", [False, True])
+    @pytest.mark.parametrize("measure_f", [qml.counts, qml.expval, qml.probs, qml.sample, qml.var])
+    @pytest.mark.parametrize(
+        "meas_obj", [qml.Y(0), [1], [1, 0], "mcm", "composite_mcm", "mcm_list"]
+    )
+    def test_simple_dynamic_circuit(self, shots, measure_f, postselect, reset, meas_obj):
```

Comment:
Instead of marking the individual tests as `unit`, you can mark the whole test class.